pkg/config/config.go

```
@@ -63,5 +63,5 @@ func createFile(name string, content []byte) error {
 	if _, err := f.Write(content); err != nil {
 		return err
 	}
-	return f.Close()
+	return nil
```

This change is wrong. It is hiding an error when closing the file after successful write. Some file systems like NFS can fail when closing a file. The current code ensures that createFile succeed only if we could close the file without an error.

The deferred close in line 62 ensures that we close the file if the write failed, in this case we want to return the error from the failed write and we don't care about a possible error from close.

---

You can check the issues. Unfortunately we have only few good first issue now.

You can try this one if you want to learn little bit about Kubernetes Go client:
https://github.com/RamenDR/ramenctl/issues/115

There are many more good first issues in the ramen project, most of them are related to ramen testing environment written in python. If you want to learn more about python this is good opportunity:
https://github.com/RamenDR/ramen/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22good%20first%20issue%22%20label%3Atest

---

TYSM